From 471e987add98b36520ece72ee493176fc7bc863c Mon Sep 17 00:00:00 2001
From: Thomas Haller <thaller@redhat.com>
Date: Fri, 21 Jan 2022 10:47:31 +0100
Subject: [PATCH] device: initialize nm_auto variable in
 _ethtool_features_reset()

The compiler is often adament to warn about maybe-uninitialized.

(cherry picked from commit 3dd854eb1ba5d13d62d890073fad1e0ec1205ea7)
---
 src/core/devices/nm-device.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/devices/nm-device.c b/src/core/devices/nm-device.c
index ea1b9647cb..ea11107c5d 100644
--- a/src/core/devices/nm-device.c
+++ b/src/core/devices/nm-device.c
@@ -1967,7 +1967,7 @@ _prop_get_x_generate_mac_address_mask(NMDevice *self, NMConnection *connection,
 static void
 _ethtool_features_reset(NMDevice *self, NMPlatform *platform, EthtoolState *ethtool_state)
 {
-    gs_free NMEthtoolFeatureStates *features;
+    gs_free NMEthtoolFeatureStates *features = NULL;
 
     features = g_steal_pointer(&ethtool_state->features);
 
-- 
GitLab

